### DISCUSSION TITLE
Docs cleanup

### DISCUSSION THREAD

**Discussion by vuki656** [MunifTanjim/nui.nvim#28] [Ideas]
Hey, big fan of the lib.

What do you think about cleaning up the docs, maybe splitting it so it's one doc per component, adding screenshots, gifs etc...

Currently it is a bit overwhelming reading the docs because it's all over the place.

I'd be willing to open a PR if you agree.

**Comment by MunifTanjim** [comment]
Yep, of course! 😃  What would be your suggestion? 

- Separate `README.md` files inside each directory
- Wiki pages
- Vim docs

🤔

**Reply by vuki656** [reply]
Hmm, I had in mind to have a readme with gifs and code snippets that could be pasted and just work and then for each component have a separate .md file with prop details and all other info.

Wiki also sounds like a good idea. Not yet sure if it's the place for these things. What do you think?

Pretty much the goal would be for someone to come, see a component they like(gif), copy paste the code snippet and it just works. Not a lot of thinking. Then if you need more details, you go to the .md file for that component.

**Reply by MunifTanjim** [reply]
Your approach sounds good. 👍🏼 Let's do what you suggested:

- The root README will contain small code snippets for components, GIF fot their results and link to the README for that specific component.
- Each component will have it's own README with details about all their options and stuffs.

PR would be very much appreciated 😃

**Reply by vuki656** [reply]
Cool, will prepare it during the next few days. Still on vacation so can't be at the computer that much.

On the off note, do you have any plans for new components? Really looking forward to new stuff and notifications.

**Reply by MunifTanjim** [reply]
> do you have any plans for new components

The existing components need some improvements (https://github.com/MunifTanjim/nui.nvim/issues/12). Also I have plan to implement scrollbar (https://github.com/MunifTanjim/nui.nvim/issues/21). Other than that I'm open to suggestions.

> Really looking forward to new stuff and notifications.

You can take a look at this in the meantime: https://github.com/rcarriga/nvim-notify

For notification in `nui.nvim` I'm thinking less about the UI and more about the notiication system. What I have in my mind is to provide a common system that can manage notifications (i.e. store, destroy, minimize, show etc.) but can work with custom UIs. So, you should be able to make your own notification UI (using `nui.popup` or `nvim-notify` etc.) on top of `nui.notification`.

**Reply by vuki656** [reply]
That's really cool regarding notifications. My only suggestion would be that you provide a basic notification that works out of the box and doesn't require much tweaking to work but then also provide some options for it to be customizable.

**Comment by vuki656** [comment]
I just started cleanup of the docs. I have some questions:

- How do you set the content of the popup? I didn't find anything in the docs
- How to make the popup automatically focused after it opens?
- I think I found another bug:

When I use this code inside a function and call it the second time, I get the following error. Maybe me manually focusing the popup and the quitting using `:wq` has something to do with it?
```
E5108: Error executing lua ...m/site/pack/packer/start/nui.nvim/lua/nui
/popup/init.lua:219: invalid border char
```

```lua
M.pop = function()
    local Popup = require("nui.popup")

    local popup = Popup({
        border = {
            style = "rounded",
            highlight = "FloatBorder",
        },
        position = "50%",
        size = {
            width = "80%",
            height = "60%",
        },
    })

    popup:mount()
end
```

**Reply by MunifTanjim** [reply]
> How do you set the content of the popup? I didn't find anything in the docs

Check this: https://github.com/MunifTanjim/nui.nvim/issues/17

> How to make the popup automatically focused after it opens?

The `enter` option (you'll find this in README.md)

---

Not sure how you're running the codes, but that should not happen. Will need to investigate. Maybe open an issue with your findings?

**Reply by vuki656** [reply]
Opened an issue about it.

Regarding text setting? What do you think about making a wrapper around NVIMs API? 

Something along the lines of `popup.set_content({...})`

**Reply by MunifTanjim** [reply]
> Regarding text setting? What do you think about making a wrapper around NVIMs API?

I don't think it would provide much value? 🤔 All the other methods are there because they do something extra than the built-in functions. For example: `set_size` does extra calculations, `map` is there because Neovim currently does not support mapping lua function, etc.

How will `set_content` differ from `nvim_buf_set_lines` or `nvim_buf_set_text`? 🤔

**Reply by vuki656** [reply]
Fair point. You are right.
